PrimeRust/solution_3/src/main.rs

```diff
@@ -0,0 +1,68 @@
+use std::time::Instant;
+
+struct PrimeSieve{
+    bits: Vec<bool>,
+    size: usize // usize means unsigned 32-bit or 64-bit integer ().
+}
+impl PrimeSieve{
+    pub(crate) fn new(n: usize) -> Self{
+        return Self{bits: vec![true; n], size: n}
+    }
+    pub(crate) fn run_sieve(&mut self){
+        let mut factor: usize = 3;
+        let q = (self.size as f32).sqrt() as usize;
+        while factor <= q { 
+            while factor < q{  // search for the  next number that is true (a prime)
+                if self.bits[factor]{
+                    break;
+                }
+                factor +=2;
+            }
+            for num in (factor*factor..=self.size).step_by(factor*2){
+                self.bits[num] = false;  // mark all multiples of this number as not prime
+            }
+            factor += 2;
+        }
+    }
+    pub(crate) fn print_results(&self, duration: f64, passes: u32){
+        eprintln!("Author; Passes; Time; Threads");  
+        println!("Blui42; {}; {}; 1", passes, duration);  // print required output to standard out
```

Comment:
Ideally, it should be this:

```suggestion
        println!("Blui42;{};{};1", passes, duration);  // print required output to standard out
```